src/workerd/jsg/promise.h

```diff
@@ -755,6 +756,17 @@ class PromiseWrapper {
       }
     }
   }
+
+private:
+  const JsgConfig config;
+
+  static bool isThenable(v8::Local<v8::Context> context, v8::Local<v8::Value> handle) {
+    if (handle->IsObject()) {
+      auto obj = handle.As<v8::Object>();
+      return check(obj->Has(context, v8StrIntern(context->GetIsolate(), "then")));
```

Comment:
Probably not worth solving in this PR, but I worry about the cost of repeatedly looking up interned strings like this all over the place. I wonder if we should be caching them somewhere, similar to how we now cache `Symbol.dispose`.

Reply:
Yep, I've been wanting to do that for a while. In Node.js we maintain a registry of handles to commonly used strings and it has a big impact. We absolutely should be caching these.